js/ui/button_group.js

```diff
@@ -0,0 +1,252 @@
+import $ from "../core/renderer";
+import Widget from "./widget/ui.widget";
+import Button from "./button";
+import CollectionWidget from "./collection/ui.collection_widget.edit";
+import registerComponent from "../core/component_registrator";
+import DataExpressionMixin from "./editor/ui.data_expression";
+import { extend } from "../core/utils/extend";
+import BindableTemplate from "./widget/bindable_template";
+import { isPlainObject } from "../core/utils/type";
+
+const BUTTON_GROUP_CLASS = "dx-buttongroup",
+    BUTTON_GROUP_WRAPPER_CLASS = "dx-buttongroup-wrapper";
+
+var ButtonCollection = CollectionWidget.inherit({
+    _initMarkup() {
+        this._updateSelectedItemKeysByIndexes(this.option("selectedIndexes"));
+        this.callBase();
+    },
+
+    _getDefaultOptions() {
+        return extend(this.callBase(), DataExpressionMixin._dataExpressionDefaultOptions(), {
+            selectedIndexes: []
+        });
+    },
+
+    _renderItemContent(options) {
+        options.container = $(options.container).parent();
+        this.callBase(options);
+    },
+
+    _focusTarget() {
```

Comment:
Part of this code looks similar to the RadioGroup. Maybe we can extract part of it to avoid the code duplication?

Reply:
My inheritor has a custom logic and contains a difference from RadioGroup.